notebook/services/contents/largefilemanager.py

```diff
@@ -0,0 +1,124 @@
+# CHANGE_FLAG: This is a new file
+from notebook.services.contents.filemanager import FileContentsManager
+from contextlib import contextmanager
+from tornado import web
+import base64
+import os, io
+
+class LargeFileManager(FileContentsManager):
+
+    def save(self, model, path=''):
+        """Save the file model and return the model with no content."""
+        chunk = model.get("chunk", None)
+        if chunk is not None and model['type'] in ['notebook', 'file']:
```

Comment:
This is a lot of new code that isn't covered - it would be good to have some testing so that this doesn't get broken accidentally later.